getTrialSeq.m

```diff
@@ -16,56 +16,165 @@
 % 5     mot_LRRL_T
 % 6     mot_RLLR_T
 
-% calculate how many trials per condition dividing them in 3 chunks to help
-% even randomization across experiment per condition
-numTrials = numTrials/3;
-
-static = (numTrials*(100-percTarget))/100/2;
-mot_LRRL = ((numTrials*(100-percTarget))/100)/4;
-mot_RLLR = ((numTrials*(100-percTarget))/100)/4;
-static_T = ((numTrials*(percTarget))/100)/2;
-mot_LRRL_T = ((numTrials*(percTarget))/100)/4;
-mot_RLLR_T = ((numTrials*(percTarget))/100)/4;
-
-% check that all of those values are integers
-if any(rem([static, mot_LRRL, mot_RLLR, static_T, mot_LRRL_T, mot_RLLR_T], 1)~=0)
-    error('that combination of number of trials and percentage of target is not possible')
-end
-
-% create the three chunks of condition and randomize them and check that: 
-% 1 - the target is not in the first trial
-% 2 - two target are not consecutive
-% 3 - there are no more them 3 same trials consecutive (less than that is impossible)
+expLength = str2num(expLength); %#ok<ST2NM>
 
-d = 1; % counter for trial number
-
-while d < numTrials*3-2
-    
-    % create a sequence of trials that contains 1 thirs of all the trials
-    trial_seq = [ ...
-        repmat(ones,1,static),  repmat(2,1,mot_LRRL),   repmat(3,1,mot_RLLR), ...
-        repmat(4,1,static_T),   repmat(5,1,mot_LRRL_T), repmat(6,1,mot_RLLR_T)];
+switch expLength
     
-    % we shuffle each chunck and concatenate them
-    trial_seq = [ Shuffle(trial_seq), Shuffle(trial_seq), Shuffle(trial_seq) ];
-   
-    % scan through the trial sequence and checks all the conditions
-    while d < length(trial_seq)-2
-        
-        if d == 1 && trial_seq(d) > 3 % no targets in the first trial
-            d = 1;
-            break
-        elseif trial_seq(d) > 3 && trial_seq(d+1) > 3 % avoid 2 consecutive targets
-            d = 1;
-            break
-            % avoid series 3 times the same conditions
-        elseif trial_seq(d) ==  trial_seq(d+1) && trial_seq(d+1) ==  trial_seq(d+2) && trial_seq(d+2) ==  trial_seq(d+3)
-            d = 1;
-            break
-        else
-            d = d+1;
+    case 1
```

Comment:
There seems to be a lot of repeated code in there: is it possible to refactor it? Maybe by creating some subfunction within this one.

Aren't case 2 and 3 just the same? Just had a quick look so I might be missing something.

Reply:
I refactored a bit and but a subfunction in the function (as we have only one function LMK if we should have a subfun folder)

Reply:
much better!

Reply:
Don't think we need a subfolder... Yet

Reply:
😂